types/relay-runtime/lib/store/RelayRecordSource.d.ts

```diff
@@ -2,7 +2,7 @@ import { MutableRecordSource, RecordMap, Record } from './RelayStoreTypes';
 import { DataID } from '../util/RelayRuntimeTypes';
 import { RecordState } from './RelayRecordState';
 
-export class RelayRecordSource implements MutableRecordSource {
+export default class RelayRecordSource implements MutableRecordSource {
```

Comment:
Looking at the source at https://github.com/facebook/relay/blob/01c68ac9a47286aec26c4706461d6c2ae46f7165/packages/relay-runtime/store/RelayRecordSource.js#L95, it appears that actually the original lib is neither using a default export nor a named one, but instead uses `module.exports = RelayRecordSource`. So unless there's something I'm unaware of, I think the way to model this in a declaration file is to have:
```ts
class RelayRecordSource { ... }
export = RelayRecordSource;
```
and then when importing:
```ts
import RecordSource = require(...);
```
(reference: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require)

Reply:
module.exports is equivalent to export default in esm

Reply:
typescript-wise, I think that's only equivalent if `esModuleInterop` is true, which we don't assume on DT because we can't be sure it will be true on other people's tsconfigs.

Reply:
I was just following conventions used in these typings. If you want this change I would recommend replacing other incorrect instances of this, For e.g.

**RelayModernStore**

https://github.com/facebook/relay/blob/805db72e807c1d085fe706497cd2ebaac7a241e9/packages/relay-runtime/store/RelayModernStore.js#L806

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/53f3d10c2298739416109c7d9a2f6d003656d8f6/types/relay-runtime/lib/store/RelayModernStore.d.ts#L21

Reply:
Ping @gabritto :)